src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -133,100 +220,100 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
-          ? isEventWithinDate(startOfSelectedDay)
-          : isEventWithinDateAndNotFinished(startOfSelectedDay)
-      )
+    const daysEvents = liveAndFutureEvents
+      .filter(isEventWithinDateAndNotFinished(day))
       .sort(eventTimeComparator)
       .map(
         prepareForSchedule({
           usersEvents: userEventIds,
         })
       );
-
     return {
       daysEvents: showPersonalisedSchedule
         ? daysEvents.filter((event) => event.isSaved)
         : daysEvents,
     };
   }, [
-    relatedVenueEvents,
+    liveAndFutureEvents,
     userEventIds,
     selectedDayIndex,
-    firstDayOfSchedule,
-    isScheduleTimeshifted,
+    minDate,
     showPersonalisedSchedule,
+    firstRangeDayTimeValue,
   ]);
 
-  // const downloadPersonalEventsCalendar = useCallback(() => {
-  //   const dayStart = addDays(startOfToday(), selectedDayIndex);
-  //   const allPersonalEvents: PersonalizedVenueEvent[] = relatedVenueEvents
-  //     .map(
-  //       prepareForSchedule({
-  //         day: dayStart,
-  //         usersEvents: userEventIds,
-  //         isForCalendarFile: true,
-  //       })
-  //     )
-  //     .filter((event) => event.isSaved);
-
-  //   downloadCalendar({
-  //     calendar: createCalendar({ events: allPersonalEvents }),
-  //     calendarName: `${PLATFORM_BRAND_NAME}_Personal`,
-  //   });
-  // }, [relatedVenueEvents, userEventIds, selectedDayIndex]);
-
-  // const downloadAllEventsCalendar = useCallback(() => {
-  //   downloadCalendar({
-  //     calendar: createCalendar({ events: relatedVenueEvents }),
-  //     calendarName: `${PLATFORM_BRAND_NAME}_Full`,
-  //   });
-  // }, [relatedVenueEvents]);
+  const downloadPersonalEventsCalendar = useCallback(() => {
+    const allPersonalEvents: PersonalizedVenueEvent[] = liveAndFutureEvents
+      .map(
+        prepareForSchedule({
+          usersEvents: userEventIds,
+        })
+      )
+      .filter((event) => event.isSaved);
+
+    downloadCalendar({
+      calendar: createCalendar({ events: allPersonalEvents }),
+      calendarName: `${PLATFORM_BRAND_NAME}_Personal`,
+    });
+  }, [userEventIds, liveAndFutureEvents]);
+
+  const downloadAllEventsCalendar = useCallback(() => {
+    downloadCalendar({
+      calendar: createCalendar({ events: liveAndFutureEvents }),
+      calendarName: `${PLATFORM_BRAND_NAME}_Full`,
+    });
+  }, [liveAndFutureEvents]);
 
   const containerClasses = classNames("NavBarSchedule", {
     "NavBarSchedule--show": isVisible,
   });
 
   return (
     <div className={containerClasses}>
-      {venueId && <ScheduleVenueDescription venueId={venueId} />}
-      {/* {!isLoadingSchedule && (
+      {/* Disabled as per designs. Up for deletion if confirmied not necessary */}
+      {/* {venueId && <ScheduleVenueDescription venueId={venueId} />} */}
+      {!isLoadingSchedule && (
         <div className="NavBarSchedule__download-buttons">
-          {hasSavedEvents && (
+          {isShowPersonalDownloadBtn ? (
```

Comment:
shouldn't we better use `&&` here?

Reply:
yep, converted it to boolean upon assignation 60ff2eb